### DISCUSSION TITLE
pymilvus2.3.x 的 Range Search 功能 怎么实现只返回l2 距离小于 0.25 的向量？

### DISCUSSION THREAD

**Discussion by ponponon** [milvus-io/milvus#26936] [Q&A and General discussion]
<img width="888" alt="图片" src="https://github.com/milvus-io/milvus/assets/38725104/712f6ac5-1794-4147-94a2-f058d7d662b4">

https://segmentfault.com/a/1190000044174364

这个例子是什么意思？完全看不懂！为什么 "radius": 10 为 ，返回的确是「该例子中就会返回距离在 10~20 之间的向量」，半径都是 10 了，为什么只返回 10-20 的向量？

我要实现，返回「距离小于 0.25 并且最多不要超过 1000个向量」这个要怎么写代码？

**Comment by ponponon** [comment]
对于 l2 距离，radius 和 range_filter 都必须要有吗？

我的需求就是，「距离小于 0.25 并且最多不要超过 1000个向量」

我只要设置 radius 或者 range_filter 中的某个值？还是两个值都需要？

我可以都设置为 0.25 吗？

<img width="868" alt="图片" src="https://github.com/milvus-io/milvus/assets/38725104/a7682c24-d33d-4256-8120-928d698dc297">


为什么这里建议 radius 要大于 range_filter ，不可以等于吗？


https://milvus.io/docs/within_range.md

**Reply by xiaofan-luan** [reply]
range filter可以不填写，且可以给limit公用
对于L2来讲，找distance < 0.25的前1000条是可以的
rangefilter的作用是找distnace < 0.25且distance > 0.1的
